File: Definitions/Game_Definitions/classes/TriggerModeHitPrereqCondition.lua
---@meta
---@diagnostic disable

---@class TriggerModeHitPrereqCondition : BaseHitPrereqCondition
---@field triggerMode gamedataTriggerMode
TriggerModeHitPrereqCondition = {}

---@return TriggerModeHitPrereqCondition
function TriggerModeHitPrereqCondition.new() return end

---@param props table
---@return TriggerModeHitPrereqCondition
function TriggerModeHitPrereqCondition.new(props) return end

---@param hitEvent gameeventsHitEvent
---@return Bool
function TriggerModeHitPrereqCondition:Evaluate(hitEvent) return end

---@param recordID TweakDBID|string
function TriggerModeHitPrereqCondition:SetData(recordID) return end

File: DecompiledGameScripts/core/gameplay/prereqs/damage/triggerModeHitPrereqCondition.swift

public class TriggerModeHitPrereqCondition extends BaseHitPrereqCondition {

  public let m_triggerMode: gamedataTriggerMode;

  public func SetData(recordID: TweakDBID) -> Void {
    this.m_triggerMode = TweakDBInterface.GetTriggerModeRecord(TweakDBInterface.GetForeignKeyDefault(recordID + t".triggerMode")).Type();
    super.SetData(recordID);
  }

  public func Evaluate(hitEvent: ref<gameHitEvent>) -> Bool {
    let result: Bool = true;
    result = Equals(hitEvent.attackData.GetTriggerMode(), this.m_triggerMode);
    return NotEquals(result, this.m_invert);
  }
}
